modules/new_serverpod_auth/serverpod_auth_session/serverpod_auth_session_flutter/lib/src/session_manager.dart

```diff
@@ -0,0 +1,155 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/foundation.dart';
+import 'package:meta/meta.dart';
+import 'package:serverpod_auth_session_client/serverpod_auth_session_client.dart';
+import 'package:serverpod_auth_session_flutter/serverpod_auth_session_flutter.dart';
+import 'package:serverpod_auth_session_flutter/src/shared_preferences_key_value_storage.dart';
+
+/// Session manager client a server using `serverpod_auth_session`.
+class SessionManager extends AuthenticationKeyManager {
+  /// Creates a new session manager instance.
+  SessionManager({
+    KeyValueStorage? storage,
+
+    /// Optional distinct secure storage for the session key (e.g. Keychain).
+    KeyValueStorage? secureStorage,
+  }) : _storage = storage ?? SharedPreferencesKeyValueStorage() {
+    _secureStorage = secureStorage ?? _storage;
+  }
+
+  final KeyValueStorage _storage;
+
+  late final KeyValueStorage _secureStorage;
+
+  final _authInfo = ValueNotifier<AuthInfo?>(null);
+
+  /// The currently logged in user.
+  ValueListenable<AuthInfo?> get authInfo => _authInfo;
+
+  String? _key;
+
+  var _init = false;
+
+  /// Read the configuration from the storage
+  Future<void> init() async {
+    if (_init) {
+      throw RepeatedSessionManagerInitError();
+    }
+    _init = true;
+
+    final sessionKey = await _secureStorage.get(
+      SessionManagerStorageKeys.sessionKeyStorageKey,
+    );
+
+    if (sessionKey != null) {
+      _key = sessionKey;
+
+      final persistedUserId = await _storage.get(
+        SessionManagerStorageKeys.authUserIdStorageKey,
+      );
+      if (persistedUserId == null) {
+        throw const IncompleteSessionManagerStorageException(
+          SessionManagerStorageKeys.authUserIdStorageKey,
+        );
+      }
+
+      final persistedScopeNames = await _storage.get(
+        SessionManagerStorageKeys.scopeNamesStorageKey,
+      );
+      if (persistedScopeNames == null) {
+        throw const IncompleteSessionManagerStorageException(
+          SessionManagerStorageKeys.scopeNamesStorageKey,
+        );
+      }
+
+      _authInfo.value = (
+        authUserId: UuidValue.fromString(persistedUserId),
+        scopeNames:
+            (jsonDecode(persistedScopeNames) as List).cast<String>().toSet(),
+      );
+    }
+  }
+
+  /// Set the current session to a logged-in user described by [authSuccess].
+  Future<void> setLoggedIn(AuthSuccess authSuccess) async {
```

Comment:
Should this name be symmetric to `logout`, ie. `login`or `setLoggedOut`?

Reply:
Absolutely makes sense.

I would probably opt for the `set*` version, as this is more a container and does not "log in" anyone.